LoopKitUI/Views/ResizeablePicker.swift

```diff
@@ -43,33 +43,41 @@ struct ResizeablePicker<SelectionValue>: UIViewRepresentable where SelectionValu
     }
 
     func updateUIView(_ view: UIPickerView, context: UIViewRepresentableContext<ResizeablePicker>) {
+        context.coordinator.updateData(newData: data)
+        view.reloadAllComponents()
         if view.selectedRow(inComponent: 0) != selectedRow {
-            view.selectRow(selectedRow, inComponent: 0, animated: true)
+            view.selectRow(selectedRow, inComponent: 0, animated: false)
         }
     }
 
     class Coordinator: NSObject, UIPickerViewDataSource, UIPickerViewDelegate {
         private var picker: ResizeablePicker
+        private var data: [SelectionValue]
```

Comment:
[nit] not exactly sure why this was necessary to fix the bouncing...?

Reply:
The old strategy was changing the id, which makes swift think there is an entirely new view here, and everything would be reconstructed. View identity loss causes a completely new UIPickerView to be created, which results in the "bouncing" animation. This update keeps the view identity intact, but has to update the available rows as they change. This is how the new available rows are passed from the new ResizeablePicker (struct) to the underlying UIPickerView.